ui/prometheus-plugin/src/model/prometheus-client.ts

```diff
@@ -83,35 +83,36 @@ export function useLabelNames(
  * values for a label.
  */
 export function useLabelValues(
-  dataSource: ResourceSelector,
+  datasource: DatasourceSelector,
   params: LabelValuesRequestParameters,
   queryOptions?: QueryOptions
 ) {
   const { labelName, ...searchParams } = params;
-  const apiUrl = `/api/v1/label/${encodeURIComponent(labelName)}/values`;
+  const apiURI = `/api/v1/label/${encodeURIComponent(labelName)}/values`;
   return useQueryWithGet<typeof searchParams, LabelValuesResponse>(
-    dataSource,
-    apiUrl,
+    datasource,
+    apiURI,
     searchParams,
     { match: 'match[]' },
     queryOptions
   );
 }
 
 function useQueryWithGet<T extends RequestParams<T>, TResponse>(
-  dataSource: ResourceSelector,
-  apiUrl: string,
+  datasourceSelector: DatasourceSelector,
+  apiURI: string,
   params: T,
   rename?: KeyNameMap<T>,
   queryOptions?: QueryOptions
 ) {
-  const { base_url: baseUrl } = usePrometheusConfig(dataSource);
-  const key = [baseUrl, apiUrl, params] as const;
+  const httpConfig = (<PrometheusSpecDatasource>usePrometheusConfig(datasourceSelector)).http;
```

Comment:
Feels like something isn't quite right if you're having to cast to `PrometheusSpecDataSource` here...

Reply:
Yup totally. But I don't know how to do it better. Probably I messed up the creation of the types/interfaces ?